### PR TITLE
[FormRecognizer] Moving tests to canadacentral; enabling AAD tests; exposing headers

### PR DESCRIPTION
**Pull request by kinelski**
A misc PR with minor changes. This PR includes:
- We're changing our live testing location to prod (canadacentral) for more stability.
- AAD is now working. Enabled AAD tests and added their recordings.
- For debugging, exposing headers and query parameters with non-sensitive data in error messages. This is something we were already doing for the old `FormRecognizer` clients.

### CODE REVIEW

File: sdk/formrecognizer/Azure.AI.FormRecognizer/src/DocumentAnalysisClientOptions.cs [hunk 1]
```diff
@@ -57,5 +58,21 @@ internal static string GetVersionString(ServiceVersion version)
...
+        /// </summary>
+        private void AddLoggedHeadersAndQueryParameters()
+        {
+            Diagnostics.LoggedHeaderNames.Add(Constants.OperationLocationHeader);
+            Diagnostics.LoggedHeaderNames.Add("apim-request-id");
+            Diagnostics.LoggedHeaderNames.Add("Strict-Transport-Security");
+            Diagnostics.LoggedHeaderNames.Add("x-content-type-options");
+            Diagnostics.LoggedHeaderNames.Add("x-envoy-upstream-service-time");
+
+            Diagnostics.LoggedQueryParameters.Add("locale");
```
Comment (maririos) [inline sdk/formrecognizer/Azure.AI.FormRecognizer/src/DocumentAnalysisClientOptions.cs:74]: should we add modelId, operationId, and resultId too?

**Review by maririos — APPROVED** [review]

File: sdk/formrecognizer/Azure.AI.FormRecognizer/src/DocumentAnalysisClientOptions.cs (hunk 1 above)
Comment (kinelski) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/src/DocumentAnalysisClientOptions.cs:74]: I could be wrong, but these IDs are not part of the query nor the headers, are they? I think they're contained in the endpoint.

Comment (maririos) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/src/DocumentAnalysisClientOptions.cs:74]: ohh you are right.
should we include the api version?

Comment (kinelski) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/src/DocumentAnalysisClientOptions.cs:74]: Added the api version.

**Pull request merged**